clang/lib/Lex/PPDirectives.cpp

```diff
@@ -2017,9 +2018,32 @@ void Preprocessor::HandleIncludeDirective(SourceLocation HashLoc,
         EnterAnnotationToken(SourceRange(HashLoc, EndLoc),
                              tok::annot_module_begin, SM);
       }
-      return;
-    }
-    if (auto *Import = std::get_if<PPCachedActions::IncludeModule>(&Include)) {
+    };
+
+    auto LoadModule = [&](const PPCachedActions::IncludeModule *Import) {
```

Comment:
Maybe factor `HandleIncludeMod` to call `LoadModule` and make the diagnostic part conditional? The error-handling logic is sufficiently interesting to not want it repeated.